network.go

```diff
@@ -47,6 +47,9 @@ type Network interface {
 
 	// EndpointByID returns the Endpoint which has the passed id. If not found, the error ErrNoSuchEndpoint is returned.
 	EndpointByID(id string) (Endpoint, error)
+
+	// EndpointByContainerID returns the Endpoint backed by a container on a network.
+	EndpointByContainerID(containerID string) (Endpoint, error)
```

Comment:
Or I would just use the Endpoint walk function at the client side, which would do the same.
Rest of the changes LGTM